version: '3.5'

services:
  inventario-fe:
    container_name: inventario-fe
    build:
      context: ./docker/nginx
      args:
        UID: $U_ID
    restart: unless-stopped
    ports:
      - "250:80"
    volumes:
      - ./public:/app/inventario/public
    depends_on:
      - inventario-be
    networks:
      - inventario-network

  inventario-be:
    container_name: inventario-be
    build:
      context: ./docker/php
      args:
        UID: $U_ID
    restart: unless-stopped
    environment:
      PHP_IDE_CONFIG: serverName=Docker
      PHP_XDEBUG_ENABLED: 1
      XDEBUG_CONFIG: remote_host=172.17.0.1 remote_port=9005 # Linux users
    #      XDEBUG_CONFIG: remote_host=host.docker.internal remote_port=9005 # macOS users
    volumes:
      - ./:/app/inventario
      - ./docker/php/xdebug-linux.ini:/usr/local/etc/php/conf.d/xdebug.ini
      #      - ./docker/php/xdebug-macos.ini:/usr/local/etc/php/conf.d/xdebug.ini
      - ~/.ssh/id_rsa:/home/inventario/.ssh/id_rsa
    depends_on:
      - inventario-db
    networks:
      - inventario-network

  inventario-db:
    container_name: inventario-db
    image: postgres:latest
    restart: unless-stopped
    ports:
      - "5432:5432"
    environment:
      POSTGRES_PASSWORD:
      POSTGRES_USER:
      POSTGRES_DB:
    volumes:
      - ./pg-data:/var/lib/postgresql/data
    networks:
      - inventario-network

volumes:
###> doctrine/doctrine-bundle ###
  database_data:
###< doctrine/doctrine-bundle ###

networks:
  inventario-network:
    external: true